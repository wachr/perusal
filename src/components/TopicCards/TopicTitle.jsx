import { h, Fragment } from "preact";
import PropTypes from "prop-types";
import Link from "@mui/material/Link";
import { Link as WouterLink } from "wouter-preact";

import {
  TOPIC_OBJECT,
  TOPIC_STRING,
  discriminating
} from "./DiscriminatingByType";

function TopicTitle({ topic }) {
  return discriminating(topic)
    .considering(TOPIC_STRING, TOPIC_OBJECT)
    .renderWith(() => (
      <div data-testid="topic-title">
        <WouterLink href={topic.topicTitle || topic}>
          <Link variant="h6">{topic.topicTitle || topic}</Link>
        </WouterLink>
      </div>
    ))
    .render();
}

TopicTitle.propTypes = {
  topic: PropTypes.oneOfType([
    PropTypes.string.isRequired,
    PropTypes.shape({
      topicTitle: PropTypes.string.isRequired
    })
  ]).isRequired
};

export default TopicTitle;
