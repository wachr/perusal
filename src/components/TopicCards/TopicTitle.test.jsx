import { h } from "preact";
import { fireEvent, render, screen } from "@testing-library/preact";

import TopicTitle from "./TopicTitle";

describe(TopicTitle.name, () => {
  it("should render objects", () => {
    render(<TopicTitle topic={{ topicTitle: "foo" }} />);
    const title = screen.queryByTestId("topic-title");
    expect(title).not.toBeNull();
    expect(title).toHaveTextContent("foo");
    expect(title).toMatchSnapshot();
  });

  it("should render strings", () => {
    render(<TopicTitle topic={"foo"} />);
    const title = screen.queryByTestId("topic-title");
    expect(title).not.toBeNull();
    expect(title).toHaveTextContent("foo");
    expect(title).toMatchSnapshot();
  });

  it("should not render arrays", () => {
    render(<TopicTitle topic={["foo"]} />);
    const title = screen.queryByTestId("topic-title");
    expect(title).toBeNull();
  });

  it("should not render objects without a title field", () => {
    render(<TopicTitle topic={{ foo: "foo" }} />);
    const title = screen.queryByTestId("topic-title");
    expect(title).toBeNull();
  });
});
