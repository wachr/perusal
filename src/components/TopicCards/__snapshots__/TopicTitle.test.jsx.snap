// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopicTitle should render objects 1`] = `
<div
  data-testid="topic-title"
>
  <a
    class="MuiTypography-root MuiTypography-h6 MuiLink-root MuiLink-underlineAlways css-igcn10-MuiTypography-root-MuiLink-root"
    href="foo"
  >
    foo
  </a>
</div>
`;

exports[`TopicTitle should render strings 1`] = `
<div
  data-testid="topic-title"
>
  <a
    class="MuiTypography-root MuiTypography-h6 MuiLink-root MuiLink-underlineAlways css-igcn10-MuiTypography-root-MuiLink-root"
    href="foo"
  >
    foo
  </a>
</div>
`;
